libportal/session.c

```diff
@@ -44,20 +44,40 @@ enum {
 
 static guint signals[LAST_SIGNAL];
 
-G_DEFINE_TYPE (XdpSession, xdp_session, G_TYPE_OBJECT)
+typedef struct {
+  GObject parent_instance;
+
+  XdpPortal *portal;
+  char *id;
+  XdpSessionType type;
+  XdpSessionState state;
+  XdpDeviceType devices;
+  GVariant *streams;
+
+  XdpPersistMode persist_mode;
+  char *restore_token;
+
+  guint signal_id;
+} XdpSessionPrivate;
+
+G_DEFINE_TYPE_WITH_PRIVATE (XdpSession, xdp_session, G_TYPE_OBJECT)
+
+static void
+session_close (XdpSession *session);
 
 static void
 xdp_session_finalize (GObject *object)
 {
   XdpSession *session = XDP_SESSION (object);
+  XdpSessionPrivate *priv = xdp_session_get_instance_private (session);
 
-  if (session->signal_id)
-    g_dbus_connection_signal_unsubscribe (session->portal->bus, session->signal_id);
+  if (priv->signal_id)
+    g_dbus_connection_signal_unsubscribe (priv->portal->bus, priv->signal_id);
 
-  g_clear_object (&session->portal);
-  g_free (session->restore_token);
-  g_free (session->id);
-  g_clear_pointer (&session->streams, g_variant_unref);
+  g_clear_object (&priv->portal);
+  g_free (priv->restore_token);
+  g_free (priv->id);
+  g_clear_pointer (&priv->streams, g_variant_unref);
```

Comment:
Could you add an extra commit to this branch changing all these cleanup free()s and whatnot to `g_clear_*()`?